MapboxDirections/MBIntersection.swift

```diff
@@ -1,54 +1,55 @@
 import Foundation
 
+/**
+ A single cross street along a step.
+ */
 @objc(MBIntersection)
 public class Intersection: NSObject, NSSecureCoding {
     
     /**
      Index into bearings/entry array.
      
-     Used to calculate the bearing just before the turn. Namely, the clockwise angle from true north to the direction of travel immediately before the maneuver/passing the intersection. Bearings are given relative to the intersection. To get the bearing in the direction of driving, the bearing has to be rotated by a value of 180. The value is not supplied for depart maneuvers.
+     The index of the item in the headings array that corresponds to the road that the containing route step uses to approach the intersection.
     */
-    public let approachIndex: Int?
+    public var approachIndex: Int = -1
     
     /**
      Index into the bearings/entry array.
      
-     Used to extract the bearing just after the turn. Namely, The clockwise angle from true north to the direction of travel immediately after the maneuver/passing the intersection. The value is not supplied for arrive maneuvers.
+     The index of the item in the headings array that corresponds to the road that the containing route step uses to leave the intersection.
     */
-    public let outletIndex: Int?
+    public var outletIndex: Int = -1
     
     /**
-     An array of booleans, corresponding in a 1:1 relationship to the bearings.
+     An array of booleans, corresponding in a 1:1 relationship to the headings.
      
      A value of true indicates that the respective road could be entered on a valid route. false indicates that the turn onto the respective road would violate a restriction.
      */
-    public var entry: [Bool]
+    public let entry: [Bool]
     
     /**
      CLLocationCoordinate2D representing the location of the intersection
     */
-    public var location: CLLocationCoordinate2D
+    public let location: CLLocationCoordinate2D
     
     /**
-     An array of CLLocationDirection values that are available at the intersection.
-     
-     The bearings describe all available roads at the intersection.
+     The geographic coordinates at the center of the intersection.
     */
     public var headings: [CLLocationDirection]
     
     /**
-     Array of Lane objects that denote the available turn lanes at the intersection.
+     Array of Lane objects.
      
-     If no lane information is available for an intersection, the lanes property will not be present.
+     If no lane information is available for an intersection, the lanes property will not be present. The first lane represents the left most lane, while the last represents the right most lane.
```

Comment:
They should line up with what's printed on the road if you were standing on it looking at the arrows